lightning/src/ln/channel.rs

```diff
@@ -2692,13 +2692,11 @@ impl<Signer: Sign> Channel<Signer> {
 	/// implicitly dropping) and the payment_hashes of HTLCs we tried to add but are dropping.
 	/// No further message handling calls may be made until a channel_reestablish dance has
 	/// completed.
-	pub fn remove_uncommitted_htlcs_and_mark_paused<L: Deref>(&mut self, logger: &L) -> Vec<(HTLCSource, PaymentHash)> where L::Target: Logger {
-		let mut outbound_drops = Vec::new();
-
+	pub fn remove_uncommitted_htlcs_and_mark_paused<L: Deref>(&mut self, logger: &L)  where L::Target: Logger {
 		assert_eq!(self.channel_state & ChannelState::ShutdownComplete as u32, 0);
 		if self.channel_state < ChannelState::FundingSent as u32 {
 			self.channel_state = ChannelState::ShutdownComplete as u32;
-			return outbound_drops;
+			return;
```

Comment:
If this is gonna be a pretty first-class use case that we expect everyone to use, should we also consider not dropping e.g. `RemoteAnnounced` HTLCs below?

Reply:
We aren't actually rejecting those HTLCs, only dropping them. If we reconnect (and the peer hasn't dropped them), then they'll re-send them.